File: app/src/main/java/seigneur/gauvain/mycourt/ui/shotDetail/TagListAdapter.java
package seigneur.gauvain.mycourt.ui.shotDetail;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import java.util.List;

import seigneur.gauvain.mycourt.R;

/**
 * RecyclerView Adapter for Shot's tags
 */
public class TagListAdapter extends RecyclerView.Adapter<TagViewHolder> {

    private List<String> tags;
    private Context context;
    public TagViewHolder tagViewHolder;

    public TagListAdapter(Context context, @NonNull List<String> tags) {
        this.context = context;
        this.tags = tags;
    }

    @Override
    public TagViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
            View view = LayoutInflater.from(parent.getContext())
                    .inflate(R.layout.list_item_tag, parent, false);
            return new TagViewHolder(view);
    }

    @Override
    public void onBindViewHolder(final TagViewHolder holder, final int position) {
        holder.tag.setText(tags.get(position));
    }

    @Override
    public int getItemCount() {
        return tags.size();
    }
}

File: app/src/main/java/seigneur/gauvain/mycourt/ui/shotDetail/tags/TagListAdapter.kt
package seigneur.gauvain.mycourt.ui.shotDetail.tags

import android.content.Context
import androidx.recyclerview.widget.RecyclerView
import android.view.LayoutInflater
import android.view.ViewGroup

import seigneur.gauvain.mycourt.R

/**
 * RecyclerView Adapter for Shot's tags
 */
class TagListAdapter(private val context: Context, private val tags: List<String>) : RecyclerView.Adapter<TagViewHolder>() {
    //var tagViewHolder: TagViewHolder? = null

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): TagViewHolder {
        val view = LayoutInflater.from(parent.context)
                .inflate(R.layout.list_item_tag, parent, false)
        return TagViewHolder(view)
    }

    override fun onBindViewHolder(holder: TagViewHolder, position: Int) {
        holder.tag.setText(tags[position])
    }

    override fun getItemCount(): Int {
        return tags.size
    }
}
